lib/CModHandler.cpp

```diff
@@ -200,20 +200,25 @@ void CIdentifierStorage::registerObject(std::string scope, std::string type, std
 std::vector<CIdentifierStorage::ObjectData> CIdentifierStorage::getPossibleIdentifiers(const ObjectCallback & request)
 {
 	std::set<std::string> allowedScopes;
+	bool found = true;
 
 	if (request.remoteScope.empty())
 	{
 		// normally ID's from all required mods, own mod and virtual "core" mod are allowed
-		if (request.localScope != "core" && request.localScope != "")
-			allowedScopes = VLC->modh->getModData(request.localScope).dependencies;
+		if(request.localScope != "core" && !request.localScope.empty())
+		{
+			allowedScopes = VLC->modh->getModDependencies(request.localScope, found);
 
+			if(!found)
+				return std::vector<ObjectData>();

```

Comment:
I see that allowedScopes always contains request.localScope and core by default. Are you sure we can not proceed with this instead of short returning here?

Reply:
`request.localScope` is invalid in this case and can not be processed. I've renamed the `found`  variable to `isValidScope`. I don't see sense to processed it for `core` scope in this case, since input data is invalid. We have an exception here now.